### ISSUE TITLE
enableUncaughtExceptionHandler causes warnings in logback configuration

### ISSUE THREAD

**Issue by HC-gcampelo** [getsentry/sentry-java#1738]
_Platform:_
- [X]  Java -> 11, no Spring
- [X]  Logback -> "io.sentry:sentry-logback:5.2.0"
- [ ]  Spring -> No

---
I have the following issue:

When creating a simple application and using sentry-logback, some messages are printed (not in the specified logback format) during initialization of Logback due to multiple getters for property 'enableUncaughtExceptionHandler' on SentryOptions.
This breaks logs ingestion, as they are expected to be in a certain format.

How to fix?
* Keep only one of [isEnableUncaughtExceptionHandler](https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryOptions.java#L1049) or [getEnableUncaughtExceptionHandler](https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryOptions.java#L1058).
* Maybe rename `isEnableUncaughtExceptionHandler` to `isUncaughtExceptionHandlerEnabled`?
I understand this is API and some libraries might not work if either are not there

**Steps to reproduce:**
- Step
- Create simple application with dependencies
```
"io.sentry:sentry-logback:5.2.0"
"net.logstash.logback:logstash-logback-encoder:6.6"
```
and `logback.xml`:
```
...
<configuration>
   ...
    <appender name="SENTRY" class="io.sentry.logback.SentryAppender">
        <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
            <level>ERROR</level>
        </filter>
        <options>
            <dsn>OVERRIDE_WITH_SENTRY_DSN_ENVIRONMENT_VARIABLE</dsn>
        </options>
    </appender>
```
on the `main`, create an slf4j logger and log something: `LoggerFactory.getLogger("isssue").info("Hi")`

**Actual result:**
- there are logs from Logback (not in logstash format) due to:
```
08:00:53,283 |-WARN in ch.qos.logback.core.joran.util.beans.BeanDescriptionFactory@36c9698e - Class 'io.sentry.SentryOptions' contains multiple getters for the same property 'enableUncaughtExceptionHandler'.
```

**Expected result:**
- No logs are printed by BeanDescriptionFactory due to multiple getters of 'enableUncaughtExceptionHandler'.

**Comment by marandaneto** [comment]
@maciejwalkowiak can you check it? thanks

**Comment by maciejwalkowiak** [comment]
Thanks @HC-gcampelo for reporting. It's only a warning message, otherwise it works as expected. 

The reason why this property has two getters was supporting setting this property through external configuration - where it can be not set - which means it must be set to `null`. 

My suggestion is to turn `enableUncaughtExceptionHandler` from `Boolean` into `boolean` (similar with other properties that have the same issue) and bind external properties to a dedicated object that allows `null` fields, then change `SentryOptions#merge(SentryOptions)` to `SentryOptions#merge(<new object>)`. Since this would be a breaking change, we could do it for 6.0.

**Comment by HC-gcampelo** [comment]
> 
> 
> Thanks @HC-gcampelo for reporting. It's only a warning message, otherwise it works as expected.
> 
> The reason why this property has two getters was supporting setting this property through external configuration - where it can be not set - which means it must be set to `null`.
> 
> My suggestion is to turn `enableUncaughtExceptionHandler` from `Boolean` into `boolean` (similar with other properties that have the same issue) and bind external properties to a dedicated object that allows `null` fields, then change `SentryOptions#merge(SentryOptions)` to `SentryOptions#merge(<new object>)`. Since this would be a breaking change, we could do it for 6.0.

Thanks @maciejwalkowiak @marandaneto ! I agree, it is a warning message, but it is caused when using the library.

As for the solution, I agree with any implementation 😅  as long as the warning message is not there anymore. Hoping to see *6.0.0* soon! 💯

**Comment by HC-gcampelo** [comment]
Hi @marandaneto , I just tested this with [6.0.0-alpha.2](https://github.com/getsentry/sentry-java/releases/tag/6.0.0-alpha.2) and it looks fixed, is this still going to be removed on the 6.0.0 release?

**Comment by marandaneto** [comment]
Its fixed by #1750
